src/magic_folder/test/cli/test_magic_folder.py

```diff
@@ -463,6 +468,75 @@ def test_poll_interval(self):
         self.assertEqual(str(ctx.exception), "--poll-interval must be a positive integer")
 
 
+class ClientEndpoint(SyncTestCase):
+    """
+    Tests related to the client-api-endpoint global config option
+    """
+
+    def setUp(self):
+        super(ClientEndpoint, self).setUp()
+        self.basedir = FilePath(self.mktemp())
+        self.nodedir = self.useFixture(
+            NodeDirectory(FilePath(self.mktemp()))
+        )
+
+    def test_convert_tcp(self):
+        """
+        a tcp: endpoint can be autoconverted
+        """
+        config = create_global_configuration(self.basedir, u"tcp:5555", self.nodedir.path)
+        self.assertThat(
+            config.api_client_endpoint,
+            Equals(u"tcp:127.0.0.1:5555")
+        )
+
+    def test_convert_tcp_host(self):
+        """
+        a tcp: endpoint can be autoconverted with host
+        """
+        config = create_global_configuration(self.basedir, u"tcp:5555:interface=127.1.2.3", self.nodedir.path)
+        self.assertThat(
+            config.api_client_endpoint,
+            Equals(u"tcp:127.1.2.3:5555")
+        )
+
+    def test_convert_fail(self):
+        """
+        unknown endpoint conversion fails
+        """
+        with ExpectedException(CannotConvertEndpointError):
+            create_global_configuration(self.basedir, u"onion:555", self.nodedir.path)
+
+    def test_convert_unix(self):
+        """
+        a tcp: endpoint can be autoconverted with host
+        """
+        config = create_global_configuration(self.basedir, u"unix:/var/run/x", self.nodedir.path)
+        self.assertThat(
+            config.api_client_endpoint,
+            Equals(u"unix:/var/run/x")
+        )
+
+    def test_set_client_endpoint(self):
+        """
+        setting the client endpoint succeeds with a valid endpoint
+        """
+        config = create_global_configuration(self.basedir, u"tcp:5555", self.nodedir.path)
+        config.api_client_endpoint = u"tcp:localhost:5555"
+        self.assertThat(
+            config.api_client_endpoint,
+            Equals(u"tcp:localhost:5555")
+        )
+
+    def test_set_invalid_client_endpoint(self):
+        """
+        setting the client endpoint fails (if it is invalid)
+        """
+        config = create_global_configuration(self.basedir, u"tcp:5555", self.nodedir.path)
+        with ExpectedException(ValueError, "Unknown endpoint type.*"):
+            config.api_client_endpoint = u"invalid:"
```

Comment:
Similarly to the above, we probably also want to know that the persisted value hasn't changed.